moveit_ros/trajectory_cache/src/trajectory_cache.cpp

```diff
@@ -0,0 +1,1340 @@
+                                                      const std::vector<geometry_msgs::msg::Pose>& waypoints,
+                                                      double step, double jump_threshold, bool avoid_collisions)
+{
+  moveit_msgs::srv::GetCartesianPath::Request out;
+
+  // Some of these parameters need us to pull PRIVATE values out of the
+  // move_group elsewhere... Yes, it is very cursed and I hate it.
+  // Fixing it requires fixing it in MoveIt.
+  moveit_msgs::msg::MotionPlanRequest tmp;
+  move_group.constructMotionPlanRequest(tmp);
+
+  out.start_state = std::move(tmp.start_state);
+  out.group_name = std::move(tmp.group_name);
+  out.max_velocity_scaling_factor = tmp.max_velocity_scaling_factor;
+  out.max_acceleration_scaling_factor = tmp.max_acceleration_scaling_factor;
+
+  out.header.frame_id = move_group.getPoseReferenceFrame();
+  out.waypoints = waypoints;
+  out.max_step = step;
+  out.jump_threshold = jump_threshold;
+  out.path_constraints = moveit_msgs::msg::Constraints();
+  out.avoid_collisions = avoid_collisions;
+  out.link_name = move_group.getEndEffectorLink();
+
+  // We were already cursed before, now we are double cursed...
+  // move_group.getNodeHandle() is UNIMPLEMENTED upstream.
+  out.header.stamp = node_->now();
+
+  return out;
+}
+
+std::vector<MessageWithMetadata<moveit_msgs::msg::RobotTrajectory>::ConstPtr>
+TrajectoryCache::fetch_all_matching_cartesian_trajectories(
+    const moveit::planning_interface::MoveGroupInterface& move_group, const std::string& move_group_namespace,
+    const moveit_msgs::srv::GetCartesianPath::Request& plan_request, double min_fraction, double start_tolerance,
+    double goal_tolerance, bool metadata_only, const std::string& sort_by)
+{
+  auto coll = db_->openCollection<moveit_msgs::msg::RobotTrajectory>("move_group_cartesian_trajectory_cache",
+                                                                     move_group_namespace);
+
+  Query::Ptr query = coll.createQuery();
+
+  bool start_ok =
+      this->extract_and_append_cartesian_trajectory_start_to_query(*query, move_group, plan_request, start_tolerance);
+  bool goal_ok =
+      this->extract_and_append_cartesian_trajectory_goal_to_query(*query, move_group, plan_request, goal_tolerance);
+
+  if (!start_ok || !goal_ok)
+  {
+    RCLCPP_ERROR(node_->get_logger(), "Could not construct cartesian trajectory query.");
+    return {};
+  }
+
+  query->appendGTE("fraction", min_fraction);
+  return coll.queryList(query, metadata_only, sort_by, true);
+}
+
+MessageWithMetadata<moveit_msgs::msg::RobotTrajectory>::ConstPtr
+TrajectoryCache::fetch_best_matching_cartesian_trajectory(
+    const moveit::planning_interface::MoveGroupInterface& move_group, const std::string& move_group_namespace,
+    const moveit_msgs::srv::GetCartesianPath::Request& plan_request, double min_fraction, double start_tolerance,
```

Comment:
Same here

Reply:
The remaining arguments are values.

Clang suggests to not do so.
https://clang.llvm.org/extra/clang-tidy/checks/readability/avoid-const-params-in-decls.html